libportal/portal.c

```diff
@@ -273,3 +280,200 @@ xdp_portal_new (void)
 {
   return g_object_new (XDP_TYPE_PORTAL, NULL);
 }
+
+/**
+ * xdp_portal_running_under_flatpak:
+ * @error: return location for a #GError pointer
+ *
+ * If you are trying to determine whether to use portals, consider using
+ * xdp_portal_running_under_sandbox() instead.
+ *
+ * Returns: %TRUE if the current process is running under a Flatpak sandbox, or
+ *   %FALSE if either an error was encountered or the process is running
```

Comment:
All of these only return an error one time, following calls will be `FALSE` without error which is unclear.

Reply:
hmm, should we just print warnings on the error code paths? That would still only be once per process but that behavior seems okay.

Reply:
I pushed a fixup that should address this.